Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2306,7 +2311,9 @@ static bool rendererContainingBlockHasMarginTrim(const RenderBox& renderer, std:
 
     if (containingBlock->isFlexibleBox()) {
         ASSERT(!marginTrimType || *marginTrimType == MarginTrimType::BlockStart);
```

Comment:
Yeah now that you bring it up this may be a bit overkill. I'm also thinking that all of these ASSERTs within this function are a bit overkill. The purpose is to return true when `box->hasTrimmedMargin()` can be called only on supported content so that it doesn't assert. Having this function ASSERT instead of just simply returning false might not be the best choice here. Do you both agree and think that they can be removed from this function? @stwrt @alanbaradlay

Reply:
My personal opinion is to remove it.

Reply:
> Yeah now that you bring it up this may be a bit overkill. I'm also thinking that all of these ASSERTs within this function are a bit overkill. The purpose is to return true when `box->hasTrimmedMargin()` can be called only on supported content so that it doesn't assert. Having this function ASSERT instead of just simply returning false might not be the best choice here. Do you both agree and think that they can be removed from this function? @stwrt @alanbaradlay

ASSERTs saved me from introducing embarrassing regressions many times in the past. Because of that I am all in favor of defensive programming and I do tend to overuse them. However I also understand that not everyone finds them useful so while I don't necessarily agree, I don't mind not having them either.